src/checkers/inference/DefaultSlotManager.java

```diff
@@ -286,8 +282,8 @@ public Slot getSlot( final AnnotationMirror annotationMirror ) {
     @Override
     public List<ConstantSlot> getConstantSlots() {
         List<ConstantSlot> constants = new ArrayList<>();
-        for (Slot slot : variables.values()) {
-            if (slot.isConstant()) {
+        for (Slot slot : slots.values()) {
+            if (!slot.isVariable()) {
```

Comment:
This should be fine if we only have variable slot vs constant slot. I just want to point out that `!slot.isVariable() != slot.isConstant()` once you have a third type of slot.  So here's my question:
Why don't we repalce these `slot.isConstant()` with `slot.getKind() == CONSTANT`, or simply modify the implementation of `slot.isConstant()`?

Reply:
slot instanceof constantslot